lib/Dialect/FIRRTL/Transforms/MemToRegOfVec.cpp

```diff
@@ -26,14 +29,45 @@ using namespace circt;
 using namespace firrtl;
 
 namespace {
+static const char excludeMemToRegClass[] =
+    "sifive.enterprise.firrtl.ExcludeMemFromMemToRegOfVec";
 struct MemToRegOfVecPass : public MemToRegOfVecBase<MemToRegOfVecPass> {
   MemToRegOfVecPass(bool replSeqMem, bool ignoreReadEnable)
       : replSeqMem(replSeqMem), ignoreReadEnable(ignoreReadEnable){};
+
   void runOnOperation() override {
-    LLVM_DEBUG(llvm::dbgs() << "\n Running MemToRegOfVecPass on module:"
-                            << getOperation().getName());
-    getOperation().getBody()->walk([&](MemOp memOp) {
+    auto circtOp = getOperation();
+    static const char dutAnnoClass[] =
+        "sifive.enterprise.firrtl.MarkDUTAnnotation";
+    DenseSet<Operation *> dutModuleSet;
+
+    auto *body = circtOp.getBody();
+
+    // Find the device under test and create a set of all modules underneath it.
+    auto it = llvm::find_if(*body, [&](Operation &op) -> bool {
+      return AnnotationSet(&op).hasAnnotation(dutAnnoClass);
```

Comment:
For future reference, we should move DUT determination into an analysis so passes don't have to be CircuitOp passes.